opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/SpanShim.java

```diff
@@ -119,17 +119,32 @@ public Span setTag(String key, Number value) {
 
   @Override
   public Span log(Map<String, ?> fields) {
-    span.addEvent(getEventNameFromFields(fields), convertToAttributes(fields));
+    ExtractedFields extractedFields = new ExtractedFields(fields);
```

Comment:
Do we need this object instead of a helper method with the logic currently in this object's constructor?

Reply:
yes, we need it, because I call it in 2 places, second is `log(long timestampMicroseconds, Map<String, ?> fields)`

Reply:
Yeah - can't you have a helper method which treats `-1` specially, for example? Just trying to avoid the allocation here which doesn't seem to be doing that much.

Reply:
done